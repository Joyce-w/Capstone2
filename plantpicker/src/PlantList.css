

.PlantList-List{
    
    column-count:  4;
    column-gap: 1em;
    margin: auto 10%;
    /* border: 1px solid black;
    height: 90vh;
    margin: auto 5%;
    display: flex;
    flex-direction: row;
    flex-wrap: wrap; */
}
.card {
    position: relative;
    background-color: rgba(238, 238, 238, 0.479);
    display: inline-block;
    margin: 0 0 1em;
    width: 100%;
}

.PlantList img {
    width: 100%;
    height: 100%;
    border-radius: 10px;
}

.plusCircle{
    position: absolute;
    color: #74B49B;
    right: 1%;
    top: 1%;
    transition: transform 1s 0s ease-in ;
}

.plusCircle:hover{
    transform: rotate(360deg);

}

