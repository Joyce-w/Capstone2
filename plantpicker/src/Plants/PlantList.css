

.PlantList-List{
    column-count:  4;
    column-gap: 1em;
    margin: auto 10%;
}
.card {
    position: relative;
    background-color: rgba(238, 238, 238, 0.479);
    display: inline-block;
    margin: 0 0 1em;
    width: 100%;
}

.PlantList img {
    width: 100%;
    height: 100%;
    border-radius: 10px;
}

.plusCircle{
    position: absolute;
    color: #74B49B;
    right: 1%;
    top: 1%;
    transition: transform 1s 0s ease-in ;
}

.plusCircle:hover{
    transform: rotate(360deg);

}

/**MEDIA QUERIES**/
@media screen and (max-width: 1024px) {
    .PlantList-List {
        column-count: 2;
        column-gap: 1em;
        margin: auto 10%;
    }
}
@media screen and (max-width: 375px) {
    .PlantList-List {
        column-count: 1;
        column-gap: 1em;
        margin: auto 10%;
    }
}