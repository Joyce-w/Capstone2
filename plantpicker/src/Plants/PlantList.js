import React, { useState, useEffect } from "react";
import { Link } from "react-router-dom";

import "./PlantList.css";
import { PlusCircle } from "phosphor-react";
import PlantsApi from "../api";




function PlantList() {

    //create state to store plant data
    const [allPlants, setAllPlants] = useState([]);
    //Get all plants data
    useEffect(() => {
        async function allPlants() {
            const res = await PlantsApi.getAllPlants();
            setAllPlants(() => res);
        }

        allPlants() 
    },[])

    return (
        <div className="PlantList">
            <h1> Category of Plants</h1>
            <div className="PlantList-List">
                
                {allPlants && allPlants.map(p =>
                    <div className="card" key={ p.id}>
                        <Link to={`/plants/${p.id}` }>
                            <div>
                                <img src={p.img} alt={ p.plant_name}></img>
                            <PlusCircle size={35} className="plusCircle"/>
                            </div>
                            
                            <div className="card-body">
                                <h3 className="card-title">{ p.plant_name}</h3>
                            </div>                        
                        </Link>

                    </div>                    
                    )
                }



            </div>
        </div>

    )
}

export default PlantList;