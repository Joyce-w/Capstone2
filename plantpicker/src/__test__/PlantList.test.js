import { render, fireEvent, screen } from "@testing-library/react";
import {MemoryRouter} from 'react-router-dom'
import PlantList from "../Plants/PlantList";

it("renders w/o crashing", () => {
    render(<MemoryRouter><PlantList/></MemoryRouter>)
})

it("matches snapshot", function () {
    const { asFragment } = render(<MemoryRouter><PlantList /></MemoryRouter>);
    expect(asFragment()).toMatchSnapshot();
})

test("if links are on the page", () => {
    const { getByText } = render(<MemoryRouter><PlantList /></MemoryRouter>);

    let h1 = getByText("Category of Plants");
    let pothos = getByText("Pothos");

    expect(h1).toBeInTheDocument();
})
