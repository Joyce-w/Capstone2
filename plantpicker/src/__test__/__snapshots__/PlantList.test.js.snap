// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="PlantList"
  >
    <h1>
       Category of Plants
    </h1>
    <div
      class="PlantList-List"
    />
  </div>
</DocumentFragment>
`;
